processor/routingprocessor/logs.go

```diff
@@ -80,32 +87,32 @@ func (p *logProcessor) route(ctx context.Context, l plog.Logs) error {
 	}{}
 	var errs error
 
-	resLogsSlice := l.ResourceLogs()
-	for i := 0; i < resLogsSlice.Len(); i++ {
-		resLogs := resLogsSlice.At(i)
-
-		attrValue := p.extractor.extractAttrFromResource(resLogs.Resource())
-		exp := p.router.defaultExporters
-		// If we have an exporter list defined for that attribute value then use it.
-		if e, ok := p.router.exporters[attrValue]; ok {
-			exp = e
-			if p.config.DropRoutingResourceAttribute {
-				resLogs.Resource().Attributes().Remove(p.config.FromAttribute)
+	for i := 0; i < l.ResourceLogs().Len(); i++ {
+		rlogs := l.ResourceLogs().At(i)
+		ltx := tqllogs.NewTransformContext(plog.LogRecord{}, pcommon.InstrumentationScope{}, rlogs.Resource())
+
+		for key, route := range p.router.routes {
+			exp := p.router.defaultExporters
+			var gKey string
+			if route.expression.Condition(ltx) {
+				route.expression.Function(ltx)
+				exp = route.exporters
+				gKey = key
 			}
-		}
 
-		if rEntry, ok := groups[attrValue]; ok {
-			resLogs.MoveTo(rEntry.resLogs.AppendEmpty())
-		} else {
-			newResLogs := plog.NewResourceLogsSlice()
-			resLogs.MoveTo(newResLogs.AppendEmpty())
-
-			groups[attrValue] = struct {
-				exporters []component.LogsExporter
-				resLogs   plog.ResourceLogsSlice
-			}{
-				exporters: exp,
-				resLogs:   newResLogs,
+			if g, ok := groups[gKey]; ok {
+				rlogs.MoveTo(g.resLogs.AppendEmpty())
+			} else {
+				newResLogs := plog.NewResourceLogsSlice()
+				rlogs.MoveTo(newResLogs.AppendEmpty())
+
+				groups[gKey] = struct {
+					exporters []component.LogsExporter
+					resLogs   plog.ResourceLogsSlice
+				}{
+					exporters: exp,
```

Comment:
If I'm reading this right, we are falling through all the conditions for the matches, right? Meaning, if I have two TQLs and both match the current data point, it's passed to both exporters. Could you double-check that we have a test asserting this behavior? Could you also positively confirm this is a new behavior, and if so, document this accordingly?

Reply:
I've extended tests to cover this behavior.

Previously, if we would specify the following routing entries: 
```
    table:
    - value: acme
      exporters: [otlp/1]
    - value: acme
      exporters: [otlp/2]    
```
which is equivalent to smth like with the support of TQL expressions
```
    table:
    - expression: route() where isMatch(..., acme) == true
      exporters: [otlp/1]
    - expression: route() where isMatch(..., .*me) == true
      exporters: [otlp/2]    
```
then it would be the same as providing the following config:
```
    table:
    - value: acme
      exporters: [otlp/1, otlp/2]    
```

So I would say that the behavior was the same, but I have explicitly documented this behavior.

Reply:
```
    table:
    - value: acme
      exporters: [otlp/1]
    - value: acme
      exporters: [otlp/2]    
```

While possible, this isn't a case I had envisioned before but you are right, this would work and send the same data to multiple exporters for the same span.

What I had in mind was a different case though:

```
    table:
    - expression: route() where isMatch(attributes.tenant, acme) == true
      exporters: [otlp/acme]
    - expression: route() where isMatch(attributes.tenant, ecorp) == true
      exporters: [otlp/ecorp]
    - expression: route() where isMatch(attributes.vip, true) == true
      exporters: [otlp/vip]    
```

In this case, we want to send each tenant-specific data point to the tenant's backend, but also send all VIPs to its own storage in addition to the tenant's. I think it's implicit that this would work, but a test would record our thoughts.